Assets/MixedRealityToolkit/Inspectors/Profiles/BaseMixedRealityProfileInspector.cs

```diff
@@ -90,12 +88,18 @@ protected static bool RenderProfile(SerializedProperty property, bool showAddBut
         /// <param name="renderProfileInBox">if true, render box around profile content, if false, don't</param>
         /// <param name="serviceType">Optional service type to limit available profile types.</param>
         /// <returns>True, if the profile changed.</returns>
-        private static bool RenderProfileInternal(SerializedProperty property, bool showAddButton, bool renderProfileInBox, Type serviceType = null)
+        private static bool RenderProfileInternal(SerializedProperty property, Type profileType,
+            bool showAddButton, bool renderProfileInBox, Type serviceType = null)
         {
             var profile = property.serializedObject.targetObject as BaseMixedRealityProfile;
             bool changed = false;
             var oldObject = property.objectReferenceValue;
 
+            if (profileType != null && !profileType.IsSubclassOf(typeof(BaseMixedRealityProfile)) && profileType != typeof(BaseMixedRealityProfile))
+            {
+                profileType = null;
+            }
```

Comment:
Is the scenario in which this gets passed an incorrect profile type something where the user dragged a non-profile asset onto the profile field on the inspector?